docs/core/metrics_datadog.md

```diff
@@ -0,0 +1,235 @@
+
+### Install
+
+???+ note
+    If you are using Datadog Forwarder, you can skip this step step.
+
+To adhere to Lambda best practices and effectively minimize the size of your development package, we recommend using the official Datadog layers built specifically for the SDK and extension components. Below is the template that demonstrates how to configure a SAM template with this information.
+
+```yaml hl_lines="13 14 22 24" title="AWS Serverless Application Model (SAM) example"
+--8<-- "examples/metrics_datadog/sam/template.yaml"
+```
+
+If you prefer not to utilize the Datadog SDK provided through the Datadog layer, add `aws-lambda-powertools[datadog]` as a dependency in your preferred tool: _e.g._, _requirements.txt_, _pyproject.toml_. This will ensure you have the required dependencies before using Datadog provider.
+
+### Creating metrics
+
+You can create metrics using `add_metric`. Optional parameter such as timestamp can be included, but if not provided, the Datadog Provider will automatically use the current timestamp by default.
+
+???+ tip
+	You can initialize DadatadogMetrics in any other module too. It'll keep track of your aggregate metrics in memory to optimize costs (one blob instead of multiples).
+
+=== "add_metrics_with_provider.py"
+
+    ```python hl_lines="6 12"
+    --8<-- "examples/metrics_datadog/src/add_metrics_with_provider.py"
+    ```
+
+=== "add_metrics_without_provider.py"
+
+    ```python hl_lines="11"
+    --8<-- "examples/metrics_datadog/src/add_metrics_without_provider.py"
+    ```
+
+???+ warning "Warning: Do not create metrics outside the handler"
+    Metrics added in the global scope will only be added during cold start. Disregard if you that's the intended behavior.
+
+### Adding tags
+
+Datadog offers the flexibility to configure tags per metric. To provider a better experience for our customers, you can pass an arbitrary number of keyword arguments (kwargs) that can be user as a tag.
+
+=== "add_metrics_with_tags.py"
+
+    ```python hl_lines="9"
+    --8<-- "examples/metrics_datadog/src/add_metrics_with_tags.py"
+    ```
+
+### Adding default tags
+
+If you want to set the same tags for all metrics, you can use the `set_default_tags` method or the `default_tags` parameter in the `log_metrics` decorator and then persist tags across the Lambda invocations.
+
+If you'd like to remove them at some point, you can use `clear_default_tags` method.
+
+???+ note
+    When default tags are configured and an additional specific tag is assigned to a metric, the metric will exclusively contain that specific tag.
+
+=== "set_default_tags.py"
+
+    ```python hl_lines="5"
+    --8<-- "examples/metrics_datadog/src/set_default_tags.py"
+    ```
+
+=== "set_default_tags_log_metrics.py"
+
+    ```python hl_lines="6 9"
+    --8<-- "examples/metrics_datadog/src/set_default_tags_log_metrics.py"
+    ```
+
+### Flushing metrics to standard output
```

Comment:
Use an alternative title like `Exporting to Datadog Log Forwarder` or something explicit to Log Forwarder.

Reason for this is that we need to avoid having two `Flush metrics` - it can confuse customers who are glancing through quickly. The more explicit the better.